src/viewer.js

```diff
@@ -3122,94 +3163,22 @@ function onCanvasKeyDown( event ) {
     this.raiseEvent('canvas-key', canvasKeyDownEventArgs);
 
     if ( !canvasKeyDownEventArgs.preventDefaultAction && !event.ctrl && !event.alt && !event.meta ) {
+
+        const code = event.originalEvent.code;
+
+        if (isNavKey(code)) {
+            this._navKeysDown[code] = true; // Mark this key as held down in the viewer's internal tracking object
+            this._navKeyFrames[code] = 0; // Reset key frames
```

Comment:
This is good; I think this is the right place for resetting the _navKeyFrames. However, onCanvasKeyDown gets called on key repeat (if the user is holding the key down), and we don't want to reset in that case. Check to see if `_navKeysDown[code]` is already true and don't reset the frames in that case.